compat/win32/fscache.c

```diff
@@ -237,7 +237,26 @@ static struct fsentry *fseentry_create_entry(struct fscache *cache,
 			     &(fse->u.s.st_mtim));
 	filetime_to_timespec((FILETIME *)&(fdata->CreationTime),
 			     &(fse->u.s.st_ctim));
-
+	if (fdata->EaSize > 0) {
+		WCHAR wpath[MAX_LONG_PATH];
+		HANDLE hnd;
+		struct stat bst;
+		if (xutftowcs_long_path(wpath, buf) >= 0 &&
+		    (hnd = CreateFileW(wpath,
```

Comment:
I fear that this is a definitive no-go. The FSCache is a compromise between correctness and speed, leaning very much in favor of speed. Adding a `CreateFileW()` for _every symlink_ is more than I am willing to add for every user, including users who have not even heard of WSL.

Reply:
the code here lies in condition if (fdata->EaSize > 0)
If we only set  wsl mode bits when a file is not a directory, not a link, and only X bit set. then at most cases, fdata->EaSize would be zero so we won't add so many calls to CreateFileW.

Reply:
I also thought that people don't have many symlinks. But they do... I did not find the ticket(s) in a quick search, but I vividly remember that we received a somewhat recent bug report where a symlink-only change caused problems (IIRC it was a slow-down introduced by https://github.com/git-for-windows/git/pull/2637).

So we _have_ to be careful here not to cause unwanted performance problems.